crates/ethereum/primitives/src/receipt.rs

```diff
@@ -152,6 +155,28 @@ impl RlpDecodableReceipt for Receipt {
     }
 }
 
+impl Decodable2718 for Receipt {
+    fn typed_decode(ty: u8, buf: &mut &[u8]) -> Eip2718Result<Self> {
+        let receipt_with_bloom = Self::rlp_decode_inner(buf, TxType::try_from(ty)?)?;
+        Ok(receipt_with_bloom.receipt)
+    }
+
+    fn fallback_decode(buf: &mut &[u8]) -> Eip2718Result<Self> {
+        let receipt_with_bloom = Self::rlp_decode_inner(buf, TxType::Legacy)?;
+        Ok(receipt_with_bloom.receipt)
+    }
+}
+
+impl Encodable2718 for Receipt {
+    fn encode_2718_len(&self) -> usize {
+        self.rlp_encoded_length_with_bloom(&self.bloom())
+    }
+
+    fn encode_2718(&self, out: &mut dyn BufMut) {
+        self.rlp_encode_with_bloom(&self.bloom(), out);
```

Comment:
this impl should use the bloom, instead we need to encode all of the fields without the bloom

basically just this without the bloom:

https://github.com/paradigmxyz/reth/blob/460f8400aa78a4cca7b74c05b435d5e1b1db43af/crates/ethereum/primitives/src/receipt.rs#L44-L50